backend/inspirehep/orcid/utils.py

```diff
@@ -160,15 +161,20 @@ def get_literature_recids_for_orcid(orcid):
         .id
     )
 
-    author_recid = (
-        db.session.query(PersistentIdentifier.pid_value)
+    author_record = (
+        db.session.query(PersistentIdentifier)
         .filter(
             PersistentIdentifier.object_type == "rec",
             PersistentIdentifier.object_uuid == author_rec_uuid,
             PersistentIdentifier.pid_type == "aut",
         )
         .one()
-        .pid_value
+    )
+
+    author_recid = (
+        author_record.pid_value
+        if not author_record.is_redirected()
+        else InspireRedirect.get_redirect(author_record).pid_value
```

Comment:
so the approach here is to follow redirects rather than rewrite recids in records?

Reply:
Yes, this is safer with "our" datamodel, It simple and do not require to fix too much in orcid module. It will also work when we add rewriting of recids in record in there future (as this is the plan as I remember). Also it works almost instantly and do not require to wait for a task to rewrite data in all records so at the end with this small change we can be sure that we will get here always proper data for orcid at least.